eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -322,6 +328,27 @@ object Router {
       case Right(rcu) => updateChannelUpdateSameSideAs(rcu.channelUpdate)
     }
   }
+  case class LocalChannel(channel: Either[PrivateChannel, PublicChannel]) {
+    val isPrivate: Boolean = channel.isLeft
+    val capacity: Satoshi = channel.fold(_.capacity, _.capacity)
+
+    /** Get our remote peer's nodeId. */
+    def getRemoteNodeId(localNodeId: PublicKey): PublicKey = channel.fold(
+      c => c.remoteNodeId,
+      c => if (localNodeId == c.ann.nodeId1) c.ann.nodeId2 else c.ann.nodeId1
+    )
+
+    /** Get our remote peer's channel_update: this is what must be used in invoice routing hints. */
+    def getRemoteUpdate(localNodeId: PublicKey): Option[ChannelUpdate] = channel.fold(
+      c => if (localNodeId == c.nodeId1) c.update_2_opt else c.update_1_opt,
+      c => if (localNodeId == c.ann.nodeId1) c.update_2_opt else c.update_1_opt
+    )
+
+    /** Create an invoice routing hint from that channel. Note that if the channel is private, the invoice will leak its existence. */
+    def toExtraHop(localNodeId: PublicKey): Option[ExtraHop] = getRemoteUpdate(localNodeId).map(remoteUpdate =>
+      ExtraHop(getRemoteNodeId(localNodeId), remoteUpdate.shortChannelId, remoteUpdate.feeBaseMsat, remoteUpdate.feeProportionalMillionths, remoteUpdate.cltvExpiryDelta)
+    )
+  }
```

Comment:
It seems to me that we could make `remoteNodeId` part of the constructor. It is a bit strange that we need to provide more info for a class named `LocalChannel` to tell which side is the local one.

Reply:
Agreed, I'll revisit that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1590/commits/273dd81593614f3323350c7ce0e3c52933841ca7, let me know what you think.

Reply:
I think it's better.